backend/src/achievements-log/achievements-log.controller.ts

```diff
@@ -0,0 +1,63 @@
+import { AchievementsLogService } from './achievements-log.service';
+import { ApiBearerAuth, ApiResponse, ApiTags } from '@nestjs/swagger';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  NotFoundException,
+  Param,
+  Post,
+} from '@nestjs/common';
+import { Public } from 'src/auth/auth.public.decorator';
+import { ResponseAchievementsLogDto } from './dto/response-achievements-log.dto';
+import { DeleteResult } from 'typeorm';
+import { CreateAchievementLogDto } from './dto/create-achievements-log.dto';
+import { HttpException, HttpStatus } from '@nestjs/common';
+@ApiBearerAuth()
+@ApiTags('achievements log')
+@Controller('achievements-log')
+export class AchievementsLogController {
+  constructor(
+    private readonly achievementsLogService: AchievementsLogService,
+  ) {}
+
+  @Public()
+  @Get()
+  async findAll(): Promise<ResponseAchievementsLogDto[]> {
+    return await this.achievementsLogService.findAll();
+  }
+
+  @Public()
+  @Get(':id')
+  @ApiResponse({ status: 404, description: 'Record not found.' })
+  async findById(@Param('id') id: string): Promise<ResponseAchievementsLogDto> {
+    const num: number = +id;
+
+    try {
+      return await this.achievementsLogService.findById(num);
```

Comment:
There is no problem with this, but you should be able to use `+id` in the findById call if you want
```suggestion
    try {
      return await this.achievementsLogService.findById(+id);
```